sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/PlaybackOnlyExtension.java

```diff
@@ -21,9 +26,24 @@ public void visitFeatureAnnotation(PlaybackOnly annotation, FeatureInfo feature)
 
     @Override
     public void visitSpecAnnotation(PlaybackOnly annotation, SpecInfo spec) {
+        validateExpiryTime(annotation);
+
         TestMode testMode = TestEnvironment.getInstance().getTestMode();
         if (testMode != TestMode.PLAYBACK) {
             spec.skip(String.format("Test ignored in %s mode", testMode));
         }
     }
+
+    private void validateExpiryTime(PlaybackOnly annotation) {
+        String expiryStr = annotation.expiryTime();
+        if ("".equals(expiryStr)) {
+            return;
+        }
+        OffsetDateTime now = OffsetDateTime.now(ZoneId.of(ZoneId.SHORT_IDS.get("PST")));
+        String nowStr = now.getYear() + "/" + String.format("%02d", now.getMonthValue()) + "/"
+            + String.format("%02d", now.getDayOfMonth());
+        if (expiryStr.compareTo(nowStr) < 0) {
```

Comment:
Consider parsing a date and comparing result
OR
validate that data provide inside the string is really a date, (i.e. I could write `@PlaybackOnly(expiryTime = "0")` to game the system).